c/cardano_lock.c

```diff
@@ -1,18 +1,300 @@
 // uncomment to enable printf in CKB-VM
 //#define CKB_C_STDLIB_PRINTF
 
-#include <stdio.h>
+#ifdef CKB_USE_SIM
+#include "sim_ckb_syscalls.h"
+#else  // CKB_USE_SIM
+#include "ckb_syscalls.h"
+#endif  // CKB_USE_SIM
+
+#include <stdbool.h>
 #include <stdint.h>
+#include <stdio.h>
+#include <string.h>
+#include <blake2b.h>
+
+#ifndef MOL2_EXIT
+#define MOL2_EXIT ckb_exit
+#endif
+
+#include "molecule/molecule_reader.h"
+#include "blockchain.h"
+#include "blockchain-api2.h"
 
-#include "nanocbor.h"
 #include "ed25519.h"
+#include "nanocbor.h"
+
+#include "ckb_consts.h"
+
+#define MAX_WITNESS_SIZE 32768
+#define BLAKE2B_BLOCK_SIZE 32
+#define ONE_BATCH_SIZE 32768
+#define SCRIPT_SIZE 32768  // 32k
+
+#define IDENTITY_SIZE 21
+#define IDENTITY_HASH_SIZE 20
```

Comment:
Change all term of `identity` to `auth`. See: https://github.com/XuJiandong/docs-bank/blob/master/omni_lock.md#auth